main.go

```diff
@@ -229,12 +235,52 @@ func visit(url *url.URL) {
 				// 30x but no Location to follow, give up.
 				return
 			}
-			log.Fatal("unable to follow redirect: %v", err)
+			log.Fatalf("unable to follow redirect: %v", err)
 		}
 		visit(loc)
 	}
 }
 
+func responseBody(response io.ReadCloser) string {
```

Comment:
Please find a better name for this, a good suggestion is to use the imperative, `readResponseBody` or `processResponse`.